edit-post/api/components/test/context.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import ReactTestRenderer from 'react-test-renderer';
+
+/**
+ * Internal dependencies
+ */
+import { PluginContextProvider, withPluginContext } from '../context';
+
+describe( 'plugin/context', () => {
+	describe( 'withPluginContext', () => {
+		it( 'passed the plugin context to the decorated component', () => {
+			const Component = ( props ) => {
+				return props.pluginContext;
+			};
+
+			const WrappedComponent = withPluginContext( Component );
+
+			const renderer = ReactTestRenderer.create(
+				<PluginContextProvider value="plugin-namespace">
+					<div>
+						<WrappedComponent />
+					</div>
+				</PluginContextProvider>
+			);
+
+			const tree = renderer.toJSON();
+
+			expect( tree.children[ 0 ] ).toEqual( 'plugin-namespace' );
```

Comment:
Gutenberg has [jest-enzyme](https://www.npmjs.com/package/jest-enzyme) bundled, it is possible to use snapshots here and `mount` helper. See: https://github.com/WordPress/gutenberg/blob/master/docs/testing-overview.md#snapshot-testing.